File: CS/WebSite/App_Code/Grid/Selection/AdvancedCheckBoxSelection/Template.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.Web.ASPxGridView;

public class CommandColumnHeaderTemplate : ITemplate {
    ASPxGridView gridView = null;
    public CommandColumnHeaderTemplate(ASPxGridView gridView) {
        this.gridView = gridView;
    }

    #region ITemplate Members

    public void InstantiateIn(Control container) {
        HtmlInputCheckBox checkBox = new HtmlInputCheckBox();
        container.Controls.Add(checkBox);
        checkBox.Attributes.Add("onclick", string.Format("{0}.SelectAllRowsOnPage(this.checked);", gridView.ClientID));
    }

    #endregion
}

File: VB/WebSite/App_Code/Grid/Selection/AdvancedCheckBoxSelection/Template.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.Web.ASPxGridView

Public Class CommandColumnHeaderTemplate
	Implements ITemplate
	Private gridView As ASPxGridView = Nothing
	Public Sub New(ByVal gridView As ASPxGridView)
		Me.gridView = gridView
	End Sub

	#Region "ITemplate Members"

	Public Sub InstantiateIn(ByVal container As Control) Implements ITemplate.InstantiateIn
		Dim checkBox As HtmlInputCheckBox = New HtmlInputCheckBox()
		container.Controls.Add(checkBox)
		checkBox.Attributes.Add("onclick", String.Format("{0}.SelectAllRowsOnPage(this.checked);", gridView.ClientID))
	End Sub

	#End Region
End Class
